lib/shoryuken/queue.rb

```diff
@@ -41,23 +38,55 @@ def receive_messages(options)
         map { |m| Message.new(client, self, m) }
     end
 
+    # Returns whether this queue is a FIFO queue or not.
+    # @return [TrueClass, FalseClass]
+    def is_fifo?
+      queue_attributes.attributes[FIFO_ATTRIBUTE] == 'true'
```

Comment:
@richseviora I got your point that we shouldn't memoize the generic `queue_attributes` as the visibility timeout can be changed after the queue creation, that was a good call. But I think we could memoize `is_fifo?`, so we don't need to perform a request for every message we send.

```ruby
def fifo?
  @_is_fifo ||= queue_attributes.attributes[FIFO_ATTRIBUTE] == 'true'
end
```

WDYT?